src/NumericalAlgorithms/Interpolation/MultiLinearSpanInterpolation.hpp

```diff
@@ -0,0 +1,397 @@
+                                                         const double x2)
+    const {
+  weight_t<2> weights;
+
+  auto index = Index<2>(findIndex(0, x1), findIndex(1, x2));
+
+  double xx, yy;
+  if constexpr (UniformSpacing) {
+    xx = (x1 - x[0][index[0]]) * inverseSpacing[0];
+    yy = (x2 - x[1][index[1]]) * inverseSpacing[1];
+  } else {
+    xx = (x1 - x[0][index[0]]) / (x[0][index[0] + 1] - x[0][index[0]]);
+    yy = (x2 - x[1][index[1]]) / (x[1][index[1] + 1] - x[1][index[1]]);
+  }
+
+  // Note: first index varies fastest
+  weights.weights = std::array<double, 4>({
+      (1. - xx) * (1. - yy),  // 00
+      xx * (1. - yy),         // 10
+      (1. - xx) * yy,         // 01
+      xx * yy,                // 11
+  });
+
+  // Compute indices
+  //
+  for (size_t j = 0; j < 2; ++j)
+    for (size_t i = 0; i < 2; ++i) {
+      auto tmp_index = Index<2>(index[0] + i, index[1] + j);
+      weights.index[i + 2 * j] = collapsed_index(tmp_index, numberOfPoints);
+    }
+
+  return weights;
+}
+
+/// Compute interpolation weights for 3D tables
+template <size_t Dimension, size_t NumberOfVariables, bool UniformSpacing>
+typename MultiLinearSpanInterpolation<Dimension, NumberOfVariables,
+                             UniformSpacing>::template weight_t<3>
+MultiLinearSpanInterpolation<Dimension, NumberOfVariables, UniformSpacing>::
+    getWeights(const double x1, const double x2, const double x3) const {
+  double xx, yy, zz;
+
+  weight_t<3> weights;
+
+  auto index = Index<3>(findIndex(0, x1), findIndex(1, x2), findIndex(2, x3));
+
+  if constexpr (UniformSpacing) {
+    xx = (x1 - x[0][index[0]]) * inverseSpacing[0];
+    yy = (x2 - x[1][index[1]]) * inverseSpacing[1];
+    zz = (x3 - x[2][index[2]]) * inverseSpacing[2];
+  } else {
+    xx = (x1 - x[0][index[0]]) / (x[0][index[0] + 1] - x[0][index[0]]);
+    yy = (x2 - x[1][index[1]]) / (x[1][index[1] + 1] - x[1][index[1]]);
+    zz = (x3 - x[2][index[2]]) / (x[2][index[2] + 1] - x[2][index[2]]);
+  }
+
+  // Note: first index varies fastest
+  weights.weights = std::array<double, 8>({
+      (1. - xx) * (1. - yy) * (1. - zz),  // 000
+      xx * (1. - yy) * (1. - zz),         // 100
+      (1. - xx) * yy * (1. - zz),         // 010
+      xx * yy * (1. - zz),                // 110
+      (1. - xx) * (1. - yy) * zz,         // 001
+      xx * (1. - yy) * zz,                // 101
+      (1. - xx) * yy * zz,                // 011
+      xx * yy * zz,                       // 111
+  });
+
+  // Compute indices
+  //
+  for (size_t k = 0; k < 2; ++k)
+    for (size_t j = 0; j < 2; ++j)
+      for (size_t i = 0; i < 2; ++i) {
+        auto tmp_index = Index<3>(index[0] + i, index[1] + j, index[2] + k);
+        weights.index[i + 2 * (j + 2 * k)] =
+            collapsed_index(tmp_index, numberOfPoints);
+      }
+
+  return weights;
+}
+
+template <size_t Dimension, size_t NumberOfVariables>
+using UniformLinearSpanInterpolation_t =
```

Comment:
no trailing `_t` and below. We use that only for `::type` in a metafunction (i.e. eager metafunctions)

Document these two type aliases? Doesn't need to be long, just enough so they show up in the dox :)